pkg/forward/forward.go

```diff
@@ -110,19 +111,14 @@ func NewInterStepDataForward(vertex *dfv1.Vertex,
 	// Add logger from parent ctx to child context.
 	isdf.ctx = logging.WithLogger(ctx, options.logger)
 
-	if isdf.opts.vertexType == dfv1.VertexTypeSource {
-		if isdf.opts.srcWatermarkPublisher == nil {
-			return nil, fmt.Errorf("failed to assign a non-nil source watermark publisher for source vertex data forwarder")
-		}
-		if isdf.opts.enableMapUdfStream {
-			return nil, fmt.Errorf("stream is not supported for source data transformer")
-		}
-	}
-
 	if isdf.opts.enableMapUdfStream && isdf.opts.readBatchSize != 1 {
 		return nil, fmt.Errorf("batch size is not 1 with UDF streaming")
 	}
 
+	if isdf.opts.vertexType == dfv1.VertexTypeSource {
```

Comment:
will this happen? we are not exposing this to users so these kinds of accidents won't happen, right?

Reply:
Right, this won't happen unless a developer passes in a source vertex type. My personal preference is to always add extra check, to be safe.